src/Cache/QueryCacheProfile.php

```diff
@@ -78,6 +78,9 @@ public function getCacheKey()
      */
     public function generateCacheKeys($sql, $params, $types, array $connectionParams = [])
     {
+        if (count($connectionParams) > 1 && array_key_exists('password', $connectionParams)) {
```

Comment:
Doing just isset would be more efficient:
```suggestion
        if (isset($connectionParams['password'])) {
```